routers/web/repo/githttp.go

```diff
@@ -195,8 +195,20 @@ func httpBase(ctx *context.Context) *serviceHandler {
 					return nil
 				}
 				if task.RepoID != repo.ID {
-					ctx.PlainText(http.StatusForbidden, "User permission denied")
-					return nil
+					actionsCfg := repo.MustGetUnit(ctx, unit.TypeActions).ActionsConfig()
+					taskRepo, err := repo_model.GetRepositoryByID(ctx, task.RepoID)
+					if err != nil {
+						ctx.ServerError("GetRepositoryByID", err)
+						return nil
+					}
+					if !actionsCfg.AccessbleFromOtherRepos || taskRepo.OwnerID != repo.OwnerID || !taskRepo.IsPrivate {
```

Comment:
This condition seems strange. If repo is public and task repo is public and they belongs to the same owner, then permission denied?

Reply:
Thanks for your review. After discussing with Lunny, I think there is a better way to implement this feature. I will update the code.